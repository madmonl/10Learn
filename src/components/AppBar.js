import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Button, IconButton, Typography, Toolbar, AppBar, Tabs, Tab, Badge } from '@material-ui/core/';
import MenuIcon from '@material-ui/icons/Menu';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import styled from 'styled-components';
import { PowerSettingsNew, Folder, Edit } from '@material-ui/icons';

// <div className={classes.container}>
const ToolbarCSS = styled(Toolbar)`
  display: flex;
  justify-content: space-between;
`;

const styles = {
  button: {
    size: 70,
    width: 100,
    fontSize: 15
  }
};

export class ButtonAppBar extends Component {
  constructor(props) {
     super(props);
  }
  logout = () => {
    if (this.props.guest) {
        this.props.dispatchGuestLogout();
    }
    else {
        this.props.startLogout();
    }
}

  render () {
    const { classes } = this.props;
    return (
        <div>
            <AppBar position="static" color="default">
              <div className="app-bar-header__theme">            
                <div className="content-container">
                  <ToolbarCSS>
                    <Typography color="textPrimary" variant="display2">
                        10Learn
                    </Typography>
                    <div className='app-bar-header'>
                      <Tabs value={0}>              
                        <Tab label={<span className="app-bar-header__tab">
                          <Edit />
                          תרגול
                        </span>} />
                        <Tab label={<span className="app-bar-header__tab">
                          <Folder />
                          מבחנים קודמים
                        </span>} />
                      </Tabs>
                      <Button className={classes.button} size='medium' onClick={this.logout}>
                        <PowerSettingsNew />
                        התנתק
                      </Button>
                    </div>          
                  </ToolbarCSS>
                </div>                
              </div>
            </AppBar>
        </div>
    );
  };
}

ButtonAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
    dispatchGuestLogout: () => dispatch(dispatchGuestLogout())
});

const mapStateToProps = (state) => ({
    guest: state.auth.guest
});

export default compose(
    withStyles(styles), 
    connect(mapStateToProps, mapDispatchToProps)
  )(ButtonAppBar);