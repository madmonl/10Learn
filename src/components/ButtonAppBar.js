import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Button, IconButton, Typography, Toolbar, AppBar, Tabs, Tab, Badge, Paper } from '@material-ui/core/';
import MenuIcon from '@material-ui/icons/Menu';
import { startLogout, dispatchGuestLogout } from '../actions/auth';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import styled from 'styled-components';
import { PowerSettingsNew, Folder, Edit } from '@material-ui/icons';
import { dispatchHeaderTabsNavigation } from '../actions/material'

// <div className={classes.container}>
const ToolbarCSS = styled(Toolbar)`
  display: flex;
  justify-content: space-between;
`;

const TypographyTABS = styled(Typography)`
  display: flex;
  justify-content: center;
  margin-right: 20px;
`;

const TypographyCSS = styled(Typography)`
  font: italic bold Georgia;
`;

const styles = theme => ({
  button: {
    size: 70,
    width: 100,
    fontSize: 15
  }
});

export class ButtonAppBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      appBarTabsIndex: 0
    }
    this.logout = this.logout.bind(this);
    this.onIndexSelect = this.onIndexSelect.bind(this);
  }
  
  logout = () => {
    if (this.props.guest) {
        this.props.dispatchGuestLogout();
    }
    else {
        this.props.startLogout();
    }
  }

  onIndexSelect = (e, index) => {
    this.props.dispatchHeaderTabsNavigation(this.props.tabs[index]);
  }

  render () {
    const { classes, tabs, currTab } = this.props;
    const currTabIndex = tabs.indexOf(currTab);
    return (
        <div>
            <AppBar position="static" color="default">
              <div className="app-bar-header__theme">            
                <div className="content-container">
                  <ToolbarCSS>
                    <TypographyCSS color="textPrimary" variant="display2">
                        10Learn
                    </TypographyCSS>
                    <div className='app-bar-header'>
                      <Tabs 
                        value={currTabIndex}
                        indicatorColor="secondary"
                        onChange={this.onIndexSelect}
                      >
                        <Tab label={<span className="app-bar-header__tab">
                          <Edit className="app-bar-header__icon" />
                          <TypographyTABS className="app-bar-header__text" color="textPrimary" variant="title">
                            תרגול
                          </TypographyTABS>
                        </span>} />
                        <Tab label={<span className="app-bar-header__tab">
                          <Folder className="app-bar-header__icon" />
                          <TypographyTABS className="app-bar-header__text" color="textPrimary" variant="title">
                            מבחנים קודמים
                          </TypographyTABS>
                        </span>} />
                      </Tabs>
                      <Button className={classes.button} size='medium' onClick={this.logout}>
                        <PowerSettingsNew />
                        התנתק
                      </Button>
                    </div>       
                  </ToolbarCSS>
                </div>                
              </div>
            </AppBar>
        </div>
    );
  };
}

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
    dispatchGuestLogout: () => dispatch(dispatchGuestLogout()),
    dispatchHeaderTabsNavigation: (currTab) => dispatch(dispatchHeaderTabsNavigation(currTab))
});

const mapStateToProps = (state) => ({
    guest: state.auth.guest,
    currTab: state.material.currTab,
    tabs: state.material.tabs
});

export default compose(
    withStyles(styles), 
    connect(mapStateToProps, mapDispatchToProps)
  )(ButtonAppBar);