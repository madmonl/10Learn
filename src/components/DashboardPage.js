import React from 'react';

const DashboardPage = () => (
    <div>
        10Learn
    </div>
);

export default DashboardPage;
