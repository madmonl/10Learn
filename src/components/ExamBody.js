import React, { Fragment, Component } from 'react';
import MathJax from 'react-mathjax';
import { dispatchChangeQuestion, dispatchMarkQuestion, dispatchSetAnswer } from '../actions/exam';
import { connect } from 'react-redux';

export class ExamBody extends Component {
  constructor (props) {
    super(props);
    this.solutionClassName = this.solutionClassName.bind(this);
    this.onSubmitQuestion = this.onSubmitQuestion.bind(this);
  }

  onSubmitQuestion = (index, answer) => {
    this.props.dispatchChangeQuestion((index + 1) % 10);
    this.props.dispatchMarkQuestion(index);
    this.props.dispatchSetAnswer(index, answer);
  } 
  
  solutionClassName = (index, questions, currQuestion, questionsStatus, answersStatus) => {
    // one of:
    // "exam__answer-item" when still in exam - clickable.
    // "exam__answer-correct" not clickable - correct solution only.
    // "exam__answer-mistake" only a mistaken solution that was chosen.
    // or exam finished and
    // "exam__answer-default" not clickable not chosen, not correct answer.
    if (questionsStatus[currQuestion] === 'being_answered') {
      return "exam__answer-item"
    } else if (questions[currQuestion].index === index) {
      return "exam__answer-correct exam__answer-item"
    } else if (questions[currQuestion].index !== index 
      && answersStatus[currQuestion] === index) {
      return "exam__answer-mistake exam__answer-item"
    } else {
      return "exam__answer-default exam__answer-item"
    };
  }
  
  render () {
    const { questions, currQuestion, questionsStatus, answersStatus } = this.props;
    
    return (
      <div>
        <div className="exam__question-navigation-container">
          <div className="exam__question">
            <MathJax.Provider>
              <div>
                  <MathJax.Node formula={questions[currQuestion].question} />
              </div>
            </MathJax.Provider>
          </div>
        </div>
        <div className="exam__answers">
          {questions[currQuestion].solutions.map((solution, index) =>
            <div 
              className={
                this.solutionClassName(index, questions, currQuestion, questionsStatus, answersStatus)
              }
              key={index}
              onClick={() => this.onSubmitQuestion(currQuestion, index)}
            >
              <MathJax.Provider>
                <div>
                    <MathJax.Node formula={solution} />
                </div>
              </MathJax.Provider>
            </div>             
          )}
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
    dispatchChangeQuestion: (question) => dispatch(dispatchChangeQuestion(question)),
    dispatchMarkQuestion: (index) => dispatch(dispatchMarkQuestion(index)),
    dispatchSetAnswer: (index, answer) => dispatch(dispatchSetAnswer(index, answer))
});

const mapStateToProps = (state) => ({
    questions: state.exam.questions,
    currQuestion: state.exam.currQuestion,
    questionsStatus: state.exam.questionsStatus,
    answersStatus: state.exam.answersStatus,
});
  
export default connect(mapStateToProps, mapDispatchToProps)(ExamBody);
