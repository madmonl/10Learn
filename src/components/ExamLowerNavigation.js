
import React, { Fragment, Component } from 'react';
import { dispatchChangeQuestion } from '../actions/exam';
import { Button } from '@material-ui/core/';
import { withStyles } from '@material-ui/core/styles';
import { Send } from '@material-ui/icons';
import { connect } from 'react-redux';
import compose from 'recompose/compose';

export const styles = theme => ({
  button: {
    margin: theme.spacing.unit
  },
  rightIcon: {
    marginLeft: theme.spacing.unit
  },
  leftIcon: {
    marginLeft: theme.spacing.unit
  }
})

export class ExamLowerNavigation extends Component {
  constructor (props) {
    super(props);
    this.onQuestionChange = this.onQuestionChange.bind(this);
  }
  
  onQuestionChange = direction => {
    if (direction === 'prev'){
      this.props.dispatchChangeQuestion( this.props.currQuestion === 0 
        ? this.props.questions.length - 1
        : this.props.currQuestion - 1)
    } else if (direction === 'next') {
      this.props.dispatchChangeQuestion((this.props.currQuestion + 1) % this.props.questions.length)
    } else {
      throw "Button direction is incorrect.";
    }
  }

  render () {
    const { questions, currQuestion, classes } = this.props;
    
    return (
      <div>
        <div className="exam__navigation-container">
          <Button 
            variant="contained" 
            color="primary" 
            className={classes.button}
            onClick={() => this.onQuestionChange('prev')}
          >
            <Send className={classes.rightIcon} />
            שאלה קודמת            
          </Button>
          <Button 
            variant="contained" 
            color="primary" 
            className={classes.button}
            onClick={() => this.onQuestionChange('next')}
          >
            שאלה הבאה
            <Send className={classes.leftIcon} />
          </Button>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
    dispatchChangeQuestion: (question) => dispatch(dispatchChangeQuestion(question))
});

const mapStateToProps = (state) => ({
    questions: state.exam.questions,
    currQuestion: state.exam.currQuestion
});
  
export default compose(
  withStyles(styles), 
  connect(mapStateToProps, mapDispatchToProps))
  (ExamLowerNavigation);

    