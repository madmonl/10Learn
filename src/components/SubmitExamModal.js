import React, { Fragment, Component } from 'react';
import { Button, Modal, Slide, Typography } from '@material-ui/core/';
import { withStyles } from '@material-ui/core/styles';
import { Done } from '@material-ui/icons';
import styled from 'styled-components';
import { dispatchChangeQuestion, dispatchChangeQuestionStatus, 
  dispatchCleanMarkedQuestions, dispatchSetGrade, startAddExam } from '../actions/exam';
import PropTypes from 'prop-types';
import compose from 'recompose/compose';
import { connect } from 'react-redux';

function getModalStyle() {
  return {
    margin:'auto'
  };
}

export const styles = theme => ({
  button: {
    margin: theme.spacing.unit
  },
  leftIcon: {
    marginLeft: theme.spacing.unit
  }
})

export class SubmitExamModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false
    }
    this.onSubmitExam = this.onSubmitExam.bind(this);
    this.onSpectateExamSolutions = this.onSpectateExamSolutions.bind(this);
    this.onModalClose = this.onModalClose.bind(this);
  }

  componentDidUpdate (prevProps) {
    if(prevProps.questionsStatus !== this.props.questionsStatus) {
      const { questions, answersStatus, questionsStatus, selectedSubjects } = this.props;
      const grade = (this.props.questionsStatus.reduce((prevQuestionsStatuses, questionStatus) => {
        if (questionStatus === 'correct') {
          prevQuestionsStatuses.push(questionStatus);
          return prevQuestionsStatuses;
        } else {
          return prevQuestionsStatuses
        }
      }, []).length / 10) * 100;
      this.props.dispatchSetGrade(grade);   
      this.props.startAddExam({
        questions,
        questionsStatus,
        answersStatus,
        selectedSubjects,
        grade
      })
    }
  }
  
  onSubmitExam = () => {
    // Open Modal
    var index = 0;
    for (; index < 10; index++) {
      this.props.dispatchChangeQuestionStatus(index, this.props.questions[index].index === this.props.answersStatus[index]
        ? 'correct'
        : 'mistake'
      );
    };
    this.setState({ open: true });
  }

  onSpectateExamSolutions = () => {
    // Clean marked questions so we can mark correct and wrong 
    // answers in red or green respectively. 
    this.props.dispatchCleanMarkedQuestions();
    this.props.dispatchChangeQuestion(0);
    this.setState({ open: false });
  }

  onModalClose = () => {
    // Close Modal
    this.setState({ open: false })
  }
  render () {
    const { questions, classes, answeredQuestions, questionsStatus, 
          answersStatus, grade } = this.props,
          { open } = this.state;
          
    return (
      <div>
        <Button 
          onClick={this.onSubmitExam} 
          variant="contained" 
          size="small" 
          className="button--submit_exam"
        >            
          סיום מבחן
          <Done className={classes.IconLeft} />
        </Button>
        <Modal
          aria-labelledby="simple-modal-title"
          aria-describedby="simple-modal-description"
          open={open}
          onClose={() => this.onModalClose()}
          className="MuiModal-root-container"
        >
          <Slide direction="up" in={open} mountOnEnter unmountOnExit>
            <div style={getModalStyle()} className={"exam__modal-box"}>
              <Typography className="exam__modal_message" variant="title" id="modal-title">
                הציון שלך הוא: {grade}
              </Typography> 
              <div className="exam__modal-buttons">               
                <Button onClick={this.onSpectateExamSolutions} variant="contained" size="small" className={classes.button}>            
                  צפייה בתשובות
                  <Done className={classes.IconLeft} />
                </Button>
                <Button onClick={() => {}} variant="contained" size="small" className={classes.button}>            
                  מבחן חדש
                  <Done className={classes.IconLeft} />
                </Button>
              </div>
            </div>
          </Slide>
        </Modal>
      </div>
    );
  };
};
    
const mapDispatchToProps = (dispatch) => ({
    dispatchChangeQuestion: (question) => dispatch(dispatchChangeQuestion(question)),
    dispatchChangeQuestionStatus: (index, status) => dispatch(dispatchChangeQuestionStatus(index, status)),
    dispatchCleanMarkedQuestions: () => dispatch(dispatchCleanMarkedQuestions()),
    dispatchSetGrade: (grade) => dispatch(dispatchSetGrade(grade)),
    startAddExam: (exam) => dispatch(startAddExam(exam))
});

const mapStateToProps = (state) => ({
    questions: state.exam.questions,
    answeredQuestions: state.exam.answeredQuestions,
    questionsStatus: state.exam.questionsStatus,
    answersStatus: state.exam.answersStatus,
    selectedSubjects: state.exam.selectedSubjects,
    grade: state.exam.grade
});

export default compose(
  withStyles(styles), 
  connect(mapStateToProps, mapDispatchToProps))
  (SubmitExamModal);

SubmitExamModal.propTypes = {
  classes: PropTypes.object.isRequired,
};