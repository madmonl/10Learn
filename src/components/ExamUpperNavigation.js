import React, { Component } from 'react';
import NumbersNavigation from './NumbersNavigation';
import SubmitExamModal from './SubmitExamModal';




export default () =>
  <div className="exam__upper-footer">
    <NumbersNavigation />
    <SubmitExamModal />
  </div>

