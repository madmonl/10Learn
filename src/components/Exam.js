import React, { Fragment, Component } from 'react';
import ExamUpperNavigation from './ExamUpperNavigation';
import ExamBody from './ExamBody';
import ExamLowerNavigation from './ExamLowerNavigation';

export default () => 
  <div className="exam">
    <ExamUpperNavigation />
    <ExamBody />
    <ExamLowerNavigation />        
  </div>

