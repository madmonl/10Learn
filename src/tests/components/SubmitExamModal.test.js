import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { SubmitExamModal } from '../../components/SubmitExamModal';
import { Modal, Button } from '@material-ui/core/';
import { Provider } from 'react-redux';
import exams from '../fixtures/exams';

let wrapper, dispatchSetGrade, dispatchChangeQuestion,
  dispatchCleanMarkedQuestions, dispatchChangeQuestionStatus, 
  startAddExam;

const createMockStore = configureMockStore([thunk]);

beforeEach(() => {
  dispatchSetGrade = jest.fn();
  dispatchChangeQuestion = jest.fn();
  dispatchCleanMarkedQuestions = jest.fn();
  dispatchChangeQuestionStatus = jest.fn();
  startAddExam = jest.fn();
  wrapper = mount(
    <SubmitExamModal
      dispatchSetGrade={dispatchSetGrade} 
      dispatchChangeQuestion={dispatchChangeQuestion} 
      dispatchCleanMarkedQuestions={dispatchCleanMarkedQuestions} 
      dispatchChangeQuestionStatus={dispatchChangeQuestionStatus}
      startAddExam={startAddExam} 
      questions={exams[0].questions}
      answeredQuestions={exams[0].answeredQuestions}
      questionsStatus={exams[0].questionsStatus}
      answersStatus={exams[0].answersStatus}
      grade={exams[0].grade}
      classes={{}}
    />
  );
});

test('should render SubmitExamModal correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should close modal', () => {
  expect(wrapper.state('open')).toBe(false)
  wrapper.setState({ open: true })
  expect(wrapper.state('open')).toBe(true)
  wrapper.find(Modal).prop('onClose')();
  expect(wrapper.state('open')).toBe(false)
})

test('should clean some exam values from store so we can spectate exam solutions', () => {
  const store = createMockStore({ 
    currQuestion: 1,
    exams: exams
  });
  wrapper = mount(shallow(
    <Provider store={store}>
      <SubmitExamModal
        dispatchSetGrade={dispatchSetGrade} 
        dispatchChangeQuestion={dispatchChangeQuestion} 
        dispatchCleanMarkedQuestions={dispatchCleanMarkedQuestions} 
        dispatchChangeQuestionStatus={dispatchChangeQuestionStatus}
        startAddExam={startAddExam} 
        questions={exams[0].questions}
        answeredQuestions={exams[0].answeredQuestions}
        questionsStatus={exams[0].questionsStatus}
        answersStatus={exams[0].answersStatus}
        grade={exams[0].grade}
        questions={exams[0].questions}
        answersStatus={exams[0].answersStatus}
        classes={{}}
      />
    </Provider>
  ).get(0));
  wrapper.setState({ open: true })
  wrapper.update();
  wrapper.find(Button).at(1).prop('onClick')();
  wrapper.update();
  expect(wrapper.state('open')).toBe(false);
  expect(dispatchCleanMarkedQuestions).toHaveBeenCalled();
  expect(dispatchChangeQuestion).toHaveBeenCalledWith(0);
});

test('testing exam submition', () => {
  wrapper.find(Button).at(0).prop('onClick')();
  expect(dispatchChangeQuestionStatus).toHaveBeenCalledTimes(10);
});

test('componentWillUpdate', () => {
  let newQuestionsStatus = [];
  var i = 0;
  for (;i < 10; i++) {
    newQuestionsStatus[i] = 'correct';
  };
  wrapper.setProps({ questionsStatus: newQuestionsStatus });
    expect(dispatchSetGrade).toHaveBeenCalledWith(100);
})