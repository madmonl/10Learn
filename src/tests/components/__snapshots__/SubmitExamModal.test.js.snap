// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SubmitExamModal correctly 1`] = `
<SubmitExamModal
  answeredQuestions={
    Array [
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
    ]
  }
  answersStatus={
    Array [
      -1,
      -1,
      -1,
      -1,
      -1,
      -1,
      -1,
      -1,
      -1,
      -1,
    ]
  }
  classes={Object {}}
  dispatchChangeQuestion={[Function]}
  dispatchChangeQuestionStatus={[Function]}
  dispatchCleanMarkedQuestions={[Function]}
  dispatchSetGrade={[Function]}
  grade={undefined}
  questions={
    Array [
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
      Object {
        "index": 1,
        "question": "1 + 2 = ?",
        "solutions": Array [
          1,
          3,
          4,
        ],
      },
    ]
  }
  questionsStatus={
    Array [
      "mistake",
      "mistake",
      "mistake",
      "mistake",
      "mistake",
      "mistake",
      "mistake",
      "mistake",
      "mistake",
      "mistake",
    ]
  }
  startAddExam={[Function]}
>
  <div>
    <WithStyles(Button)
      className="button--submit_exam"
      onClick={[Function]}
      size="small"
      variant="contained"
    >
      <Button
        className="button--submit_exam"
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit-22",
            "contained": "MuiButton-contained-12",
            "containedPrimary": "MuiButton-containedPrimary-13",
            "containedSecondary": "MuiButton-containedSecondary-14",
            "disabled": "MuiButton-disabled-21",
            "extendedFab": "MuiButton-extendedFab-19",
            "fab": "MuiButton-fab-18",
            "flat": "MuiButton-flat-6",
            "flatPrimary": "MuiButton-flatPrimary-7",
            "flatSecondary": "MuiButton-flatSecondary-8",
            "focusVisible": "MuiButton-focusVisible-20",
            "fullWidth": "MuiButton-fullWidth-26",
            "label": "MuiButton-label-2",
            "mini": "MuiButton-mini-23",
            "outlined": "MuiButton-outlined-9",
            "outlinedPrimary": "MuiButton-outlinedPrimary-10",
            "outlinedSecondary": "MuiButton-outlinedSecondary-11",
            "raised": "MuiButton-raised-15",
            "raisedPrimary": "MuiButton-raisedPrimary-16",
            "raisedSecondary": "MuiButton-raisedSecondary-17",
            "root": "MuiButton-root-1",
            "sizeLarge": "MuiButton-sizeLarge-25",
            "sizeSmall": "MuiButton-sizeSmall-24",
            "text": "MuiButton-text-3",
            "textPrimary": "MuiButton-textPrimary-4",
            "textSecondary": "MuiButton-textSecondary-5",
          }
        }
        color="default"
        component="button"
        disableFocusRipple={false}
        disabled={false}
        fullWidth={false}
        mini={false}
        onClick={[Function]}
        size="small"
        type="button"
        variant="contained"
      >
        <WithStyles(ButtonBase)
          className="MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15 MuiButton-sizeSmall-24 button--submit_exam"
          component="button"
          disabled={false}
          focusRipple={true}
          focusVisibleClassName="MuiButton-focusVisible-20"
          onClick={[Function]}
          type="button"
        >
          <ButtonBase
            centerRipple={false}
            className="MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15 MuiButton-sizeSmall-24 button--submit_exam"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-28",
                "focusVisible": "MuiButtonBase-focusVisible-29",
                "root": "MuiButtonBase-root-27",
              }
            }
            component="button"
            disableRipple={false}
            disableTouchRipple={false}
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-20"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <button
              className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15 MuiButton-sizeSmall-24 button--submit_exam"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-2"
              >
                סיום מבחן
                <pure(DoneIcon)
                  className={undefined}
                >
                  <DoneIcon
                    className={undefined}
                  >
                    <WithStyles(SvgIcon)
                      className={undefined}
                    >
                      <SvgIcon
                        className={undefined}
                        classes={
                          Object {
                            "colorAction": "MuiSvgIcon-colorAction-33",
                            "colorDisabled": "MuiSvgIcon-colorDisabled-35",
                            "colorError": "MuiSvgIcon-colorError-34",
                            "colorPrimary": "MuiSvgIcon-colorPrimary-31",
                            "colorSecondary": "MuiSvgIcon-colorSecondary-32",
                            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit-36",
                            "root": "MuiSvgIcon-root-30",
                          }
                        }
                        color="inherit"
                        component="svg"
                        fontSize="default"
                        viewBox="0 0 24 24"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root-30"
                          color={undefined}
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                          <path
                            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                          />
                        </svg>
                      </SvgIcon>
                    </WithStyles(SvgIcon)>
                  </DoneIcon>
                </pure(DoneIcon)>
              </span>
              <WithStyles(TouchRipple)
                center={false}
                innerRef={[Function]}
              >
                <TouchRipple
                  center={false}
                  classes={
                    Object {
                      "child": "MuiTouchRipple-child-41",
                      "childLeaving": "MuiTouchRipple-childLeaving-42",
                      "childPulsate": "MuiTouchRipple-childPulsate-43",
                      "ripple": "MuiTouchRipple-ripple-38",
                      "ripplePulsate": "MuiTouchRipple-ripplePulsate-40",
                      "rippleVisible": "MuiTouchRipple-rippleVisible-39",
                      "root": "MuiTouchRipple-root-37",
                    }
                  }
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    className="MuiTouchRipple-root-37"
                    component="span"
                    enter={true}
                    exit={true}
                  >
                    <span
                      className="MuiTouchRipple-root-37"
                    />
                  </TransitionGroup>
                </TouchRipple>
              </WithStyles(TouchRipple)>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </Button>
    </WithStyles(Button)>
    <WithStyles(Modal)
      aria-describedby="simple-modal-description"
      aria-labelledby="simple-modal-title"
      className="MuiModal-root-container"
      onClose={[Function]}
      open={false}
    >
      <Modal
        BackdropComponent={[Function]}
        aria-describedby="simple-modal-description"
        aria-labelledby="simple-modal-title"
        className="MuiModal-root-container"
        classes={
          Object {
            "hidden": "MuiModal-hidden-45",
            "root": "MuiModal-root-44",
          }
        }
        disableAutoFocus={false}
        disableBackdropClick={false}
        disableEnforceFocus={false}
        disableEscapeKeyDown={false}
        disablePortal={false}
        disableRestoreFocus={false}
        hideBackdrop={false}
        keepMounted={false}
        manager={
          ModalManager {
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "modals": Array [],
          }
        }
        onClose={[Function]}
        open={false}
      />
    </WithStyles(Modal)>
  </div>
</SubmitExamModal>
`;
